src/com/homework/lesson10/Main.java

```diff
@@ -0,0 +1,76 @@
+package com.homework.lesson10;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class Main {
+    static long startCount2=0;
+    public static void main(String[] args) throws InterruptedException {
+        List<String> https = new CopyOnWriteArrayList<>();
+        https.add("https://habr.com/");
+        https.add("https://github.com/");
+        https.add("https://www.google.com/");
+        long start = System.currentTimeMillis();
+        long startCount = 0;
+        for (String http : https) {
+            try (InputStream inputStream = new URL(http).openStream()) {
```

Comment:
If  .openStream returns not buffered stream (I think so, and there is no mention of it in the documentation), you can wrap inputStream in BufferedInputStream here and below, it will be faster

Reply:
Is it ok, if I will change like:
BufferedInputStream inputStream = (BufferedInputStream) new URL(https.get(0)).openStream()
or BufferedInputStream has any method to convert InputStream?

Reply:
oh, I find this code
BufferedInputStream inputStream = new BufferedInputStream(new URL(http).openStream())
is it right?

Reply:
yap